components/badge.jsx

```diff
@@ -0,0 +1,62 @@
+var React       = require('react'),
+    ImageTag    = require('./imagetag.jsx');
+
+var Badge = React.createClass({
+    propTypes: {
+        title: React.PropTypes.string.isRequired,
+        status: React.PropTypes.string,
+        icon: React.PropTypes.string.isRequired,
+        icon2x: React.PropTypes.string.isRequired
+    },
+    render: function (){
+
+        /**
+         * returning the Status HTML
+         * @param statusVal
+         * @returns ReactElement
+         */
+        function status (statusVal) {
+            var classNames = 'label label-default ' + statusVal;
+            var statusIcon = "";
+
+            if (statusVal === undefined) {
+                return "";
+            }
+
+            switch (statusVal) {
```

Comment:
we're in React, so there's no need to build HTML using strings: we can use JS and JSX to get the job done:

```
var classnames = require("classnames");
...
var statusIcons = {
  pending: 'fa-clock-o',
  achieved: 'fa-check',
  eligible: 'fa-pencil'
};
var statusClasses = classnames('fa', 'fa-fw', statusIcons[statusVal]);
var statusIcon = <i className={statusClasses} />;

return (
  <div className="status">
    <div className={classNames}>
      <span>{ statusIcon }</span>{ statusVal }
    </div>
  </div>
);
```